db/migrate/20150504075221_populate_participating.rb

```diff
@@ -0,0 +1,16 @@
+class PopulateParticipating < ActiveRecord::Migration
+  def up
+    progress_bar = ProgressBar.create( format: "(\e[32m%c/%C\e[0m) %a |%B| \e[31m%e\e[0m ",
+                                       progress_mark: "\e[32m/\e[0m",
+                                       total: Discussion.count )
+    Discussion.reset_column_information
```

Comment:
Okay, so in plaintext:

For each discussion
Each discussion reader for that discussion, which has a user that either
a) has an event in discussion.events
b) is the author of the discussion
Shall be set to participating in the discussion.

This took ~20min on my machine running a recent prod dump.